### PR TITLE
APEXMALHAR-2234 FilterOperator documentation

### PR DESCRIPTION
**Pull request by yogidevendra**

### CODE REVIEW

**Comment by yogidevendra** [comment]
@chinmaykolhatkar Could you please review this?

File: docs/operators/filter.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Filter
+=============
+
+## Operator Objective
+This operator receives an POJO ([Plain Old Java Object](https://en.wikipedia.org/wiki/Plain_Old_Java_Object)) as an incoming tuple
+and based on the filter condition it shall emit the filtered POJO on one output port and the rest on another output port.
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: Can you reword it as follows:
"...it emits filtered tuples on one output port and rest on another output port"

File: docs/operators/filter.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Filter operator supports quasi Java expressions to specify filter rule.
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md:8]: Might not be for this review. But we need a documentation on quasi-Java expressions.

File: docs/operators/filter.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+Filter operator does not hold any state and is **idempotent**, **fault-tolerant** and **statically/dynamically partitionable**.
+
+## Operator Usecase
+1. ***Customer data*** usually contains a field customer category/segment. One wants some analysis to be done for specific customer segment. One could filter the records based on segment.
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: "One could use this filter operator to filter the records based on segment for some analysis for specific customer segment."

File: docs/operators/filter.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+2. ***Log data*** processing pipeline may want to filter logs from specific machine/router/switch.
+
+## Operator Information
+1. Operator location: ***malhar-library***
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: Would it make sense to make this a link to : https://github.com/apache/apex-malhar/tree/master/library

File: docs/operators/filter.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+3. Operator state: ***Evolving***
+3. Java Packages:
+    * Operator: ***[com.datatorrent.lib.filter.FilterOperator](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/filter/FilterOperator.html)***
+
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of FilterOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *condition* | condition/expression with which Filtering is done. | String | Yes | N/A |
+| *additionalExpressionFunctions* | List of import classes/method that should be made statically available to expression to use. | `List< String>`| No | Empty List |
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: List<String> there is a space between < & S

File: docs/operators/filter.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+### Platform Attributes that influences operator behavior
+| **Attribute** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *port.input.attr.TUPLE_CLASS* | TUPLE_CLASS attribute on input port indicates the class of POJO which incoming tuple | Class or FQCN| Yes |
+
+
+### Ports
+| **Port** | **Description** | **Type** | **Mandatory** |
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: Instead of "Mandatory", can we make the tuple as "Connection Required"?

File: docs/operators/filter.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+| -------- | ----------- | ---- | ------------------ |
+| *input* | Tuple which needs to be filtered are received on this port | Object (POJO) | Yes |
+| *truePort* | Tuples which satisfies `condition` are emitted on this port | Object (POJO) | No |
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: link condition to "props"

File: docs/operators/filter.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+| *falsePort* | Tuples which does not satisfy `condition` are emitted on this port | Object (POJO) | No |
```
Comment (chinmaykolhatkar) [inline docs/operators/filter.md]: link "condition" to props.

**Review by chinmaykolhatkar — CHANGES_REQUESTED** [review]
Can you please take care of comments and squash and rebase with master?

**Comment by chinmaykolhatkar** [comment]
Please add link to mkdocs.yml for the link to this document to appear in navigation.

File: docs/operators/filter.md (hunk 2 above)
Comment (yogidevendra) [reply docs/operators/filter.md:8]: APEXMALHAR-2266 created for this

File: docs/operators/filter.md (hunk 1 above)
Comment (yogidevendra) [reply docs/operators/filter.md]: done

File: docs/operators/filter.md (hunk 4 above)
Comment (yogidevendra) [reply docs/operators/filter.md]: done

File: docs/operators/filter.md (hunk 5 above)
Comment (yogidevendra) [reply docs/operators/filter.md]: done

File: docs/operators/filter.md (hunk 7 above)
Comment (yogidevendra) [reply docs/operators/filter.md]: done

File: docs/operators/filter.md (hunk 8 above)
Comment (yogidevendra) [reply docs/operators/filter.md]: done

**Comment by yogidevendra** [comment]
@chinmaykolhatkar Incorporated review comments.

**Review by chinmaykolhatkar — APPROVED** [review]
Changes looks good.

**Comment by chinmaykolhatkar** [comment]
@yogidevendra Please rebase with master. I'll merge it then.

**Comment by chinmaykolhatkar** [comment]
Travis CI is passing but Jenkins build is failing. I would like to merge this PR as the change are only related to docs folder. This change does not impact CI builds.
Though, I'll wait for one more day to merge for anyone has any objection to it. If no objection, I'll merge the PR tomorrow morning India time.

**Pull request merged**